### DISCUSSION TITLE
主键和插件是怎么实现在线获取和安装的

### DISCUSSION THREAD

**Discussion by uummxx** [the1812/Bilibili-Evolved#4670] [提问 / Q&A]
我也在写一个小插件
我也想像你这样 约定一个规则，然后方便开发
但是我看你的插件数据好像都没有存储本地，那安装组件和插件的数据时直接加到 油猴的脚本里面吗？
我也搜索了相关关键字 属实没看到是怎么解决的 
感觉好高级的样子

**Comment by the1812** [comment]
用 [GM_setValue](https://www.tampermonkey.net/documentation.php#api:GM_setValue).
实现在 src/core/settings/index.ts
